src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieRedirectContext.cs

```diff
@@ -20,15 +20,25 @@ public class CookieRedirectContext : BaseCookieContext
         /// <param name="options">The cookie handler options</param>
         /// <param name="redirectUri">The initial redirect URI</param>
         /// <param name="properties">The <see cref="AuthenticationProperties"/>.</param>
-        public CookieRedirectContext(HttpContext context, AuthenticationScheme scheme, CookieAuthenticationOptions options, string redirectUri, AuthenticationProperties properties)
-            : base(context, scheme, options, properties)
+        public CookieRedirectContext(
+            HttpContext context,
+            AuthenticationScheme scheme,
+            CookieAuthenticationOptions options,
+            AuthenticationProperties properties,
+            string redirectUri)
+            : base(context, scheme, options)
         {
+            Properties = properties;
             RedirectUri = redirectUri;
         }
 
         /// <summary>
         /// Gets or Sets the URI used for the redirect operation.
         /// </summary>
         public string RedirectUri { get; set; }
+
+        public bool Skipped { get; private set; }
```

Comment:
Yeah, the cookies handler is inconsistent with the other handlers and doesn't offer a way to skip the challenge operation. This PR looks like the perfect occasion to fix that.

I added this method/property when removing the base context class (where `Skip` was defined) and adding the missing checks in the handler + a few unit tests to check the new behavior is on my todo-list.